File: src/java/magma/PlaceholderRule.java
package magma;

import java.util.Optional;

public record PlaceholderRule(StringRule rule) {
    static String wrap(final String input) {
        return "/*" + input.replace("/*", "start").replace("*/", "end") + "*/";
    }

    Optional<String> generate(final Node node) {
        return this.rule.generate(node).map(PlaceholderRule::wrap);
    }
}

File: src/node/magma/PlaceholderRule.ts
/*import java.util.Optional;*/
/*public record PlaceholderRule(Rule rule) implements Rule {
	static String generatePlaceholder(final String input) {
		return "start" + input.replace("start", "start").replace("end", "end") + "end";
	}

	@Override
	public Optional<String> generate(final MapNode node) {
		return this.rule().generate(node).map(PlaceholderRule::generatePlaceholder);
	}
}*/
